tests/unit/backend/corpora/api_server/curator/collection/test_curation_collections.py

```diff
@@ -471,6 +471,43 @@ def test__get_private_collection__OK(self):
         self.assertEqual(200, res.status_code)
         self.assertEqual("test_collection_id_revision", res.json["id"])
 
+    def test__get_collection_with_dataset_failing_validation(self):
+        collection = self.generate_collection(
+            self.session,
+            visibility=CollectionVisibility.PRIVATE.name,
+        )
+        dataset = self.generate_dataset(
+            self.session,
+            collection=collection,
+            processing_status={
+                "processing_status": ProcessingStatus.FAILURE,
+                "validation_status": ValidationStatus.INVALID,
+                "validation_message": "test message",
+            },
+        )
+        res = self.app.get(f"/curation/v1/collections/{collection.id}")
+        self.assertEqual("FAILURE", res.json["processing_status"])
+        for resp_dataset in res.json["datasets"]:
+            if dataset.id == resp_dataset["id"]:
+                self.assertEqual("VALIDATION_FAILURE", resp_dataset["processing_status"])
+                self.assertEqual("test message", resp_dataset["processing_status_detail"])
+                break
+
+    def test__get_collection_with_dataset_failing_pipeline(self):
+        collection = self.generate_collection(
+            self.session,
+            visibility=CollectionVisibility.PRIVATE.name,
+        )
+        dataset = self.generate_dataset(
+            self.session, collection=collection, processing_status={"processing_status": ProcessingStatus.FAILURE}
+        )
+        res = self.app.get(f"/curation/v1/collections/{collection.id}")
+        self.assertEqual("FAILURE", res.json["processing_status"])
+        for resp_dataset in res.json["datasets"]:
+            if dataset.id == resp_dataset["id"]:
```

Comment:
I'm curious on your thoughts about [my strategy for verifying that conditional testing statements are triggered](https://github.com/chanzuckerberg/single-cell-data-portal/blob/68248993a6a587d7fda8675d9c1e1565ae953d47/tests/unit/backend/corpora/api_server/curator/collection/test_curation_collections.py#L143-L181). Without this sort of strategy, errors in the tested code could cause a conditional such as the one you have here to never get triggered, thus allowing the test to silently pass when it should actually fail.

Reply:
True, though in this case there should only be one dataset so I can remove the loop altogether.